samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1512 @@
+}
+
+void SparseImage::setup_camera()
+{
+	camera.type = vkb::CameraType::FirstPerson;
+	camera.set_perspective(fov_degrees, static_cast<float>(width) / static_cast<float>(height), 0.1f, 512.0f);
+	camera.set_rotation(glm::vec3(0.0f, 0.0f, 0.0f));
+	camera.set_translation(glm::vec3(0.0f, 0.0f, -50.0f));
+	camera.translation_speed = 3.0f;
+}
+
+/**
+ * 	@brief Copy a single-page, raw-pixel-data from the CPU memory into the stagging buffer.
+ */
+void SparseImage::copy_single_raw_data_block(uint8_t buffer[], const VkExtent2D blockDim, VkOffset2D offset, size_t stride)
+{
+	for (size_t row = 0U; row < blockDim.height; row++)
+	{
+		size_t position = (row + offset.y) * stride + offset.x * 4U;
+		memcpy(&buffer[row * blockDim.width * 4U], &virtual_texture.raw_data_image->get_data()[position], blockDim.width * 4U);
+	}
+}
+
+/**
+ * 	@brief Fill up the information on how the sparse image should be bound and call vkQueueBindSparse, based on the bind_list.
+ */
+void SparseImage::bind_sparse_image()
+{
+	virtual_texture.bind_list.sort();
+
+	for (size_t count = 0U; count < virtual_texture.num_pages; count++)
+	{
+		if (virtual_texture.bind_list.empty() == true)
+		{
+			virtual_texture.sparse_image_memory_bind[count].memory = VK_NULL_HANDLE;
+			continue;
+		}
+
+		size_t page_index = virtual_texture.bind_list.front();
+		if (count != page_index)
+		{
+			virtual_texture.sparse_image_memory_bind[count].memory = VK_NULL_HANDLE;
+			continue;
+		}
+
+		virtual_texture.bind_list.pop_front();
+		if (virtual_texture.page_table[page_index].bound == true)
+		{
+			continue;
+		}
+
+		if (virtual_texture.available_memory_index_list.empty() == true)
+		{
+			throw std::runtime_error("bind_sparse_image(): There is not enough memory available");
+		}
+
+		size_t memory_block_index = virtual_texture.available_memory_index_list.front();
+		virtual_texture.available_memory_index_list.pop_front();
+
+		size_t pageOffset = virtual_texture.page_size * memory_block_index;
+
+		virtual_texture.page_table[page_index].memory_index = memory_block_index;
+		virtual_texture.page_table[page_index].bound        = true;
+
+		virtual_texture.sparse_image_memory_bind[count].memory       = virtual_texture.texture_memory;
+		virtual_texture.sparse_image_memory_bind[count].memoryOffset = pageOffset;
+	}
+
+	VkBindSparseInfo bind_sparse_info = vkb::initializers::bind_sparse_info();
+	bind_sparse_info.bufferBindCount  = 0U;
+	bind_sparse_info.pBufferBinds     = nullptr;
+
+	VkSparseImageMemoryBindInfo sparse_image_memory_bind_info{};
+	sparse_image_memory_bind_info.image     = virtual_texture.texture_image;
+	sparse_image_memory_bind_info.bindCount = virtual_texture.sparse_image_memory_bind.size();
+	sparse_image_memory_bind_info.pBinds    = virtual_texture.sparse_image_memory_bind.data();
+
+	bind_sparse_info.imageBindCount = 1U;
+	bind_sparse_info.pImageBinds    = &sparse_image_memory_bind_info;
+
+	VkSparseImageOpaqueMemoryBindInfo sparse_image_opaque_memory_bind_info{};
```

Comment:
`sparse_image_opaque_memory_bind_info` is never used?

Reply:
No, it is not used. An `opaque` memory binding is used for handling the `mip_tail`. `Mip_tail` is the set of mip levels that have a smaller size than the sparse block size in bytes, all of them grouped together into an opaque region.
Currently I'm using 5 LODs for the application. Because of image's resolution, handling an mip_tail region is not required yet.

Reply:
Just potentially. I don't think it is currently required. ~could be required if we decide to change image / recalculate mip-levels for a wider spread etc.